File: backend/src/main/java/com/backend/entity/Course.java
package com.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "course")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ElementCollection
    @CollectionTable(name = "course_tags", joinColumns = @JoinColumn(name = "course_id"))
    @Column(name = "tag")
    private List<String> tags;

    @Enumerated(EnumType.STRING)
    private Level level;

    /**
     * 创建者用户 id（参考 User 实体 id）
     */
    private Long authorId;

    /**
     * syllabus 可存为 JSON 字符串或 markdown 文本（最小化实现）
     */
    @Column(columnDefinition = "TEXT")
    private String syllabus;

    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
    }

    public enum Level {
        BEGINNER,
        INTERMEDIATE,
        ADVANCED
    }
}

File: frontend/src/types/course.ts
/**
 * 课程相关类型定义
 */
export interface Course {
  id: number;
  title: string;
  description: string;
  tags: string[];
  level: "BEGINNER" | "INTERMEDIATE" | "ADVANCED";
  authorId: number;
  createdAt: string;
}

export interface UserCourse {
  courseId: number;
  courseTitle: string | null;
  isCurrentStudying: boolean;
  addedAt: string;
  studyingStartedAt: string | null;
}
